### DISCUSSION TITLE
Multiarch catchall discussion

### DISCUSSION THREAD

**Discussion by Diatrus** [ProcursusTeam/Procursus#889] [General]
Use this thread to talk about multilib progress and suggestions. Cc/ @CRKatri

**Comment by CRKatri** [comment]
If someone would like to help with this I would be very grateful if someone could go through and add the `Multi-Arch` key to every `.control`, most of it would be directly copied from Debian, and could probably be easily scripted.

**Reply by Diatrus** [reply]
I'd prefer we not do that and add it in the PACK function.

**Reply by CRKatri** [reply]
I don't see why, what benefits would that give? Adding it to the control files in simple enough.

**Comment by CRKatri** [comment]
I just squashed my `ck/multiarch-experiment` branch to `ck/multiarch` and split all the changes to `profile.d` into `633d021ffabcd580895ccee02b9fb197d1057a9c` if we want to make the transition as painless as possible in the future I would recommend cherry-picking `633d021ffabcd580895ccee02b9fb197d1057a9c` and pushing that to the APT repo so that the infrastructure is in place before anything requires it. But that's just a suggestion.

**Reply by Diatrus** [reply]
A better idea than the fallback path is to have everything related to this consolidated under a var (MULTILIB_ENABLED?) and switched based off that. So iOS 14 and under can stay the same and 15 and over can be multilib (if you got it ready in time)

**Reply by CRKatri** [reply]
However, then any bin that links procursus libraries that isn't part of procursus has to be recompiled to link the multiarch path or it won't work, because it won't find the lib in `/usr/lib` and won't know to check the multiarch directory.

**Comment by CRKatri** [comment]
I have my first multiarch bootstrap, this is compiled for iOS 14. You can download it here: https://cameronkatri.com/bootstrap-multiarch.tar.zst, **DO NOT INSTALL ON A MAIN DEVICE**. This is **completely untested** because I don't have any test phones available and I don't want to jailbreak my iPad. Please send me the output when you run a command that is not distributed as part of the bootstrap but links libraries from it. It should use the `DYLD_LIBRARY_FALLBACK_PATH` and still find the library even though it's in `/usr/lib/arm64-apple-ios/`, if it doesn't then we have a big issue.

**Reply by m1stadev** [reply]
Tried testing this on an iPhone 6S+ running iOS 14.6, but it doesn't look like there's any `awk` package, which is required by `add-shell` (and with that, the `prep_bootstrap.sh` script). Could you add that into a bootstrap please?

**Reply by CRKatri** [reply]
At some point `no-awk.patch` stopped being applied, just open `/usr/sbin/add-shell` in a text editor and change `awk` to `cat`, I'm in the process of packaging a new strap right now.

**Reply by CRKatri** [reply]
[https://cameronkatri.com/bootstrap-multiarch.tar.zst{,.asc}](https://cameronkatri.com/bootstrap-multiarch.tar.zst) should have debianutils patched. @m1stadev

**Reply by CRKatri** [reply]
@DaOfficialRolex ^^^

**Reply by m1stadev** [reply]
Sorry for the late response; I haven't had time to test the bootstrap recently until now. Upon first bootstrapping, `zsh` works, along with tools such as `dpkg`, `ncurses-bin`, `snaputil`, `coreutils`, but `apt` is missing liblz4, which doesn't look like it's included in the bootstrap.

**Reply by CRKatri** [reply]
Does `apt` work if you download https://apt.procurs.us/pool/main/iphoneos-arm64/1700/liblz4-dev_1.9.2-2_iphoneos-arm.deb and install it with `dpkg`?

**Comment by CRKatri** [comment]
I don't believe it's necessary to update every single package to be multiarch at once, I think it would be exceptable to just get the main bootstrap then multiarch the rest of the packages as they are updated. `/etc/profile.d/multiarch.sh` should take care of the linking to old or new paths in binaries.
